buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -267,6 +267,7 @@ public ByteBuf ensureWritable(int minWritableBytes) {
     }
 
     private void ensureWritable0(int minWritableBytes) {
+        ensureAccessible();
```

Comment:
@johnou you thinks that it's better to move the call of `ensureAccessible` from `ensureWritable0` to `ensureWritable`?